crates/precompile/src/lib.rs

```diff
@@ -300,33 +300,94 @@ impl Precompiles {
     }
 }
 
-/// Precompile with address and function.
+/// Unique precompile identifier.
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub enum PrecompileId {
+    /// Elliptic curve digital signature algorithm (ECDSA) public key recovery function.
+    EcRec,
+    /// SHA2-256 hash function.
+    Sha256,
+    /// RIPEMD-160 hash function.
+    Ripemd160,
+    /// Identity precompile.
+    Identity,
+    ///	Arbitrary-precision exponentiation under modulo.
+    ModExp,
+    /// Point addition (ADD) on the elliptic curve 'alt_bn128'.
+    Bn128Add,
+    /// Scalar multiplication (MUL) on the elliptic curve 'alt_bn128'.
+    Bn128Mul,
+    /// Bilinear function on groups on the elliptic curve 'alt_bn128'.
+    Bn128Pairing,
+    /// Compression function F used in the BLAKE2 cryptographic hashing algorithm.
+    Blake2F,
+    /// Verify p(z) = y given commitment that corresponds to the polynomial p(x) and a KZG proof. Also verify that the provided commitment matches the provided versioned_hash.
+    KzgPointEvaluation,
+    /// Point addition in G1 (curve over base prime field).
+    Bls12G1Add,
+    /// Multi-scalar-multiplication (MSM) in G1 (curve over base prime field).
+    Bls12G1Msm,
+    /// Point addition in G2 (curve over quadratic extension of the base prime field).
+    Bls12G2Add,
+    /// Multi-scalar-multiplication (MSM) in G2 (curve over quadratic extension of the base prime field).
+    Bls12G2Msm,
+    /// Pairing operations between a set of pairs of (G1, G2) points.
+    Bls12Pairing,
+    /// Base field element mapping into the G1 point.
+    Bls12MapFpToGp1,
+    /// Extension field element mapping into the G2 point.
+    Bls12MapFp2ToGp2,
+    /// ECDSA signature verification over the secp256r1 elliptic curve (also known as P-256 or prime256v1).
+    P256Verify,
+    /// Custom precompile identifier.
+    Custom(String),
```

Comment:
this should be `Other(Cow<'static, str>)`

and a helper fn to create a static one